src/System.Management.Automation/engine/InitialSessionState.cs

```diff
@@ -4534,14 +4534,37 @@ static InitialSessionState()
             builtinVariables.Add(
                 new SessionStateVariableEntry(
                     SpecialVariables.NativeArgumentPassing,
-                    Platform.IsWindows ? NativeArgumentPassingStyle.Windows : NativeArgumentPassingStyle.Standard,
+                    GetPassingStyle(),
                     RunspaceInit.NativeCommandArgumentPassingDescription,
                     ScopedItemOptions.None,
                     new ArgumentTypeConverterAttribute(typeof(NativeArgumentPassingStyle))));
 
             BuiltInVariables = builtinVariables.ToArray();
         }
 
+        /// <summary>
+        /// Assigns the default behavior for native argument passing.
+        /// If the system is non-Windows, we will return Standard.
+        /// If the experimental feature is enabled, we will return Windows.
+        /// Otherwise, we will return Legacy.
+        /// </summary>
+        private static NativeArgumentPassingStyle GetPassingStyle()
+        {
+            // If non-Windows, use the standard passing style
+            if (!Platform.IsWindows)
+            {
+                return NativeArgumentPassingStyle.Standard;
+            }
+
+            // If the experimental feature is enabled return Windows..
+            if (ExperimentalFeature.IsEnabled(ExperimentalFeature.PSWindowsNativeCommandArgPassing))
+            {
+                return NativeArgumentPassingStyle.Windows;
+            }
+
+            return NativeArgumentPassingStyle.Legacy;
```

Comment:
Please use `#if UNIX` guard here too.
```suggestion
#if UNIX
            return NativeArgumentPassingStyle.Standard;
#else
            if (ExperimentalFeature.IsEnabled(ExperimentalFeature.PSWindowsNativeCommandArgPassing))
            {
                return NativeArgumentPassingStyle.Windows;
            }

            return NativeArgumentPassingStyle.Legacy;
#endif
```